packages/graphile-build-pg/res/introspection-query.sql

```diff
@@ -102,7 +102,7 @@ with
       (pg_catalog.pg_relation_is_updatable(rel.oid, true)::bit(8) & B'00000100') = B'00000100' as "isDeletable"
     from
       pg_catalog.pg_class as rel
-      left join pg_catalog.pg_description as dsc on dsc.objoid = rel.oid and dsc.objsubid = 0
+      left join pg_catalog.pg_description as dsc on dsc.objoid = rel.oid and dsc.objsubid = 0 and dsc.classoid = (select oid from pg_catalog.pg_class where relname = 'pg_class')
```

Comment:
Here's the comments/sources I get in one of my databases:

```
# select classoid, count(*), relname from pg_description inner join pg_class on (classoid = pg_class.oid) group by classoid, relname order by count desc;
 classoid | count |    relname
----------+-------+----------------
     1255 |  2878 | pg_proc
     2617 |   787 | pg_operator
     2607 |   132 | pg_conversion
     1247 |    73 | pg_type
     3602 |    16 | pg_ts_config
     3600 |    16 | pg_ts_dict
     2601 |     6 | pg_am
     3764 |     5 | pg_ts_template
     2612 |     4 | pg_language
     3079 |     4 | pg_extension
     2615 |     3 | pg_namespace
     3456 |     3 | pg_collation
     1259 |     2 | pg_class
     3601 |     1 | pg_ts_parser
(14 rows)
```

Seems legit to me that when we want comments on a class we'd only want ones from `pg_class`. I'd definitely be interested to see where your stray comments came from though?

Reply:
Feels to me we should replicate this check on columns (also `pg_class`), procs (`pg_proc` instead of `pg_class`), etc.

Reply:
Filed as issue: https://github.com/graphile/graphile-build/issues/216

Reply:
I have no idea actually. They came into existence after upgrading with `pg_upgrade`, so perhaps there's either a bug or a quirk in how that works. It doesn't technically seem "wrong" to have multiple rows though, the schema indeed supports it, but unexpected yes.